janus_server/src/trace.rs

```diff
@@ -50,6 +64,43 @@ pub struct TokioConsoleConfiguration {
     pub listen_address: Option<SocketAddr>,
 }
 
+/// Selection of an exporter for OpenTelemetry spans.
+#[derive(Clone, Debug, PartialEq, Eq, Serialize, Deserialize)]
+pub enum OpenTelemetryTraceConfiguration {
+    #[serde(rename = "jaeger")]
+    Jaeger(JaegerConfiguration),
+    #[serde(rename = "otlp")]
+    Otlp(OtlpConfiguration),
+}
+
+impl OpenTelemetryTraceConfiguration {
+    pub fn is_enabled(&self) -> bool {
+        match self {
+            OpenTelemetryTraceConfiguration::Jaeger(config) => config.enabled,
+            OpenTelemetryTraceConfiguration::Otlp(config) => config.enabled,
+        }
+    }
+}
+
+#[derive(Clone, Debug, Default, PartialEq, Eq, Serialize, Deserialize)]
+pub struct JaegerConfiguration {
+    /// Enable the Jaeger exporter.
+    pub enabled: bool,
```

Comment:
Is there a good reason not to use the existence of the Jaeger/OTLP config as the enabled flag? That is, is there a use-case for setting this `enabled` flag to false?

If not, I suggest dropping this flag & `OtlpConfiguration.enabled`, and using the following semantics:
* If `open_telemetry_config` is omitted entirely, disabled Jaeger/OTLP.
* If `open_telemetry_config.jaeger` is specified, enable Jaeger.
* If `open_telemetry_config.otlp` is specified, enable OTLP.